examples/react/apollo-client-defer/src/App.tsx

```diff
@@ -0,0 +1,56 @@
+import { useQuery } from '@apollo/client';
+
+import { useFragment, graphql, FragmentType, isFragmentReady } from './gql';
+import { SlowAndFastFieldWithDeferQuery } from './gql/graphql';
+
+export const slowFieldFragment = graphql(/* GraphQL */ `
+  fragment SlowFieldFragment on Query {
+    slowField(waitFor: 5000)
+  }
+`);
+
+const alphabetQuery = graphql(/* GraphQL */ `
+  query SlowAndFastFieldWithDefer {
+    fastField
+    ...SlowFieldFragment @defer
+
+    ... @defer {
+      inlinedSlowField: slowField(waitFor: 5000)
+    }
+  }
+`);
+
+const SlowDataField = (props: { data: FragmentType<typeof slowFieldFragment> }) => {
+  const data = useFragment(slowFieldFragment, props.data);
+  return <p>{data.slowField}</p>;
+};
+
+const InlinedSlowDataField = (props: { data: SlowAndFastFieldWithDeferQuery }) => {
+  try {
+    // @ts-expect-error - this field should be either undefined or a string
+    const _ = props.data.inlinedSlowField.toLowerCase();
+  } catch (e) {}
+
+  if (!props.data.inlinedSlowField) {
+    return null;
+  }
+  return <p>{props.data.inlinedSlowField} inlined fragment</p>;
+};
+
+function App() {
+  const { data } = useQuery(alphabetQuery);
+
+  return (
+    <div className="App">
+      {data && (
+        <>
+          <p>{data.fastField}</p>
+          {isFragmentReady(alphabetQuery, slowFieldFragment, data) && <SlowDataField data={data} />}
+          <InlinedSlowDataField data={data} />
+        </>
+      )}
+    </div>
+  );
+}
```

Comment:
I just tried this and it did not raise a TypeScript error. 🤔  Any idea how we can make this an error?

<img width="777" alt="Screenshot 2023-04-04 at 10 25 01" src="https://user-images.githubusercontent.com/14338007/229733352-da8cf66d-4261-4234-94f0-8910180487ba.png">

Reply:
`isFragmentReady` is a runtime function, so it won't render `SlowFieldData` in the runtime. And since it's a type guard, it makes `foo` a proper type. Also, I guess passing an empty object (foo) is valid because if it's deferred, it actually can be empty, no?
<img width="515" alt="CleanShot 2023-04-04 at 11 12 36@2x" src="https://user-images.githubusercontent.com/9019397/229745301-4c4f90fc-c520-4b92-872b-4c24acb82cd6.png">

Reply:
Yeah, i am just wondering whether we should be stricter and only allow passing actual query stuff that has a `" $fragmentsRefs"` field defined in the type defs 🤔

Reply:
Ok I think I found a way to make it more typesafe:
```
data: FragmentType<TypedDocumentNode<Incremental<TFrag>, any>> | null | undefined
```
And it will be `TypedDocumentString` in the string mode

Reply:
So, for example if I pass an invalid object (`organization` is invalid, `project` would be valid):
 
<img width="335" alt="CleanShot 2023-04-06 at 17 15 13@2x" src="https://user-images.githubusercontent.com/9019397/230423273-b6c2db97-1bbe-466d-aea3-aaf4b59d1468.png">

I'll see get the following error:
<img width="638" alt="CleanShot 2023-04-06 at 17 15 30@2x" src="https://user-images.githubusercontent.com/9019397/230423471-a108192f-6869-40b2-8abf-14ec9b5a8b1c.png">